### PR TITLE
Fix approx_distinct for TIMESTAMP inputs

### LINKED ISSUE #8163: The result of approx_distinct is different from Presto when the data type is Timestamp
### Bug description

approx_distinct(Timestamp) in Velox doesn't match Presto: 303 vs. 300.

Presto uses the millisecond value of Timestamp to calculate the hash value, while Velox uses the Timestamp itself to calculate the hash value.

The fix is to use the millisecond value of the Timestamp type in Velox to calculate the hash value, consistent with Presto.

In Presto
```
SELECT approx_distinct(from_unixtime(c0, 'UTC'), 0.023)
FROM (
  VALUES 
(905746043),
(215588628),
(804489195),
(123780566),
(1547661049),
(1474116657),
(676849069),
(242395458),
(1663259494),
(6172523),
(1502828344),
(1165264382),
(389916048),
(1489969255),
(590028214),
(1530268049),
(1442140895),
(1101068119),
(18564324),
(1348580750),
(987304073),
(846890662),
(1086013123),
(673360407),
(711313738),
(1505910931),
(1277164609),
(776157290),
(1048000710),
(660705326),
(505691714),
(961828890),
(537569964),
(1647153315),
(1108509062),
(520390042),
(1562168178),
(228138641),
(216429757),
(144986243),
(180040370),
(110553666),
(1454402814),
(1486942531),
(997177933),
(871037064),
(308541818),
(711722272),
(986972089),
(215701966),
(152925788),
(366321659),
(113202996),
(1247813013),
(546682678),
(1037620761),
(1259331611),
(346821716),
(448349379),
(856678905),
(1588465889),
(27989033),
(1039987365),
(38011096),
(548543874),
(1404831961),
(975542822),
(959676592),
(1608754451),
(267101901),
(213545088),
(411402585),
(1289797454),
(719489171),
(1368220405),
(1616706430),
(828203515),
(1570089093),
(1628274222),
(485133624),
(27427879),
(990773112),
(1459971704),
(1028681149),
(739003497),
(1118473283),
(1565971490),
(1068799967),
(1146617959),
(1618569154),
(665063458),
(1097264699),
(508254755),
(854529091),
(1219877122),
(1242473274),
(994825085),
(139569085),
(1526186858),
(1171040845),
(818924697),
(821791397),
(348825393),
(800751100),
(993949416),
(1606702195),
(204924001),
(316434687),
(1161599775),
(1556431645),
(1118134980),
(370288917),
(68683355),
(194139295),
(805980244),
(818881098),
(360451089),
(1208001474),
(52798147),
(598388005),
(1697617591),
(754341415),
(359118509),
(1050351834),
(669223073),
(593933215),
(376515894),
(1234323405),
(636220801),
(844276664),
(468877292),
(1231620784),
(1569192080),
(1134676627),
(1544360014),
(1276701796),
(1589856582),
(814313893),
(1680863679),
(739917942),
(765373060),
(1053592948),
(591298237),
(152894038),
(122274444),
(623608432),
(1339595245),
(572295711),
(1321711446),
(590461652),
(1664387047),
(428889160),
(1377810728),
(278253551),
(961302697),
(1680351042),
(432479133),
(305100482),
(121681876),
(162676687),
(1386185849),
(296454414),
(1085901027),
(476072718),
(1440824721),
(361407928),
(634185558),
(1053217842),
(690789996),
(594467035),
(479000962),
(1370697389),
(534973573),
(573382020),
(92837967),
(1316242771),
(1230911977),
(493448656),
(996693522),
(624579281),
(1183702549),
(1485447212),
(1018416963),
(1697783451),
(565498027),
(303127780),
(663167997),
(1452032688),
(85813656),
(1240964020),
(1017695886),
(709395319),
(1685280659),
(90614567),
(80436851),
(238123582),
(952174127),
(1233525955),
(581688080),
(659899384),
(1689732617),
(652322190),
(1110302747),
(1118499653),
(613260577),
(532033743),
(1574560991),
(712631959),
(1557403197),
(1192713312),
(1301296158),
(244186605),
(1044350190),
(181831942),
(1290912745),
(822040398),
(1304588205),
(1662835346),
(1515325990),
(1571678450),
(1614670680),
(36245749),
(604608314),
(994733406),
(457432767),
(928114984),
(1301598474),
(1401630896),
(333023883),
(536581422),
(1492901206),
(1073520903),
(867502031),
(139580427),
(1020543630),
(548771218),
(14093829),
(355143596),
(1072109805),
(1666000817),
(948929112),
(795580571),
(845239387),
(1457143076),
(1131797345),
(689576653),
(718274990),
(58358977),
(189340222),
(1020981950),
(56573081),
(1170279140),
(341385133),
(1016435808),
(392540472),
(1473363502),
(1503271797),
(742883730),
(1126268774),
(102760974),
(6724123),
(1128844321),
(364765074),
(640712296),
(1271970136),
(1376912164),
(904079677),
(155738804),
(938159854),
(754081924),
(170977612),
(1245648272),
(594199549),
(93928477),
(383693436),
(49327638),
(1240835868),
(1472760961),
(810811291),
(39680907),
(1306166790),
(1132650540),
(1074308364),
(264514521),
(122149830),
(502527729),
(789859647),
(1257340037),
(902424944),
(495109640),
(641843161),
(814744535),
(1654231877),
(1093485783),
(256026031),
(1646443316),
(864493601),
(888890381),
(313820154),
(140549411),
(264514521),
(14093829),
(814744535),
(711313738),
(1459971704),
(1304588205),
(348825393),
(952174127),
(1290912745),
(139569085),
(376515894),
(1290912745),
(1526186858),
(1473363502),
(814313893),
(1565971490),
(1664387047),
(194139295),
(1208001474),
(1376912164),
(624579281),
(814744535),
(1028681149),
(1368220405),
(305100482),
(1618569154),
(1128844321),
(593933215),
(392540472),
(1557403197),
(1097264699),
(623608432),
(392540472),
(1557403197),
(814744535),
(1017695886),
(1503271797),
(305100482),
(1588465889),
(1530268049),
(305100482),
(1557403197),
(93928477),
(476072718),
(1569192080),
(493448656),
(1440824721),
(1662835346),
(305100482),
(305100482),
(1485447212),
(1072109805),
(1108509062),
(256026031),
(624579281),
(789859647),
(181831942),
(181831942),
(1440824721),
(1503271797),
(994825085),
(1037620761),
(123780566),
(411402585),
(376515894),
(1183702549),
(110553666),
(1068799967),
(1565971490),
(256026031),
(110553666),
(536581422),
(1662835346),
(1321711446),
(573382020),
(548771218),
(845239387),
(1028681149),
(1473363502),
(476072718),
(244186605),
(565498027),
(1473363502),
(6724123),
(1108509062),
(355143596),
(1277164609),
(1530268049),
(1473363502),
(495109640),
(536581422),
(14093829),
(624579281),
(1565971490),
(85813656),
(1440824721),
(14093829),
(155738804),
(1230911977),
(613260577),
(493448656),
(905746043),
(1662835346),
(613260577),
(1085901027),
(1037620761),
(1085901027),
(814744535),
(1697617591),
(1053217842),
(1588465889),
(1459971704),
(305100482),
(313820154),
(1680863679),
(1074308364),
(1321711446),
(659899384),
(613260577),
(162676687),
(1685280659),
(1289797454),
(123780566),
(264514521),
(1370697389),
(457432767),
(1680351042),
(110553666),
(170977612),
(14093829),
(1697617591),
(457432767),
(122149830),
(1271970136),
(242395458),
(948929112),
(194139295),
(1565971490),
(1128844321),
(1233525955),
(1502828344),
(1454402814),
(361407928),
(1044350190),
(305100482),
(1037620761),
(1588465889),
(432479133),
(1259331611),
(493448656)
) AS tmp (c0);
```

The result is 300.

In Velox

```
TEST_F(ApproxDistinctTest, globalAggTimeStamp) {
  auto mills = makeFlatVector<int64_t>(
      {905746043000,  215588628000,  804489195000,  123780566000,
       1547661049000, 1474116657000, 676849069000,  242395458000,
       1663259494000, 6172523000,    1502828344000, 1165264382000,
       389916048000,  1489969255000, 590028214000,  1530268049000,
       1442140895000, 1101068119000, 18564324000,   1348580750000,
       987304073000,  846890662000,  1086013123000, 673360407000,
       711313738000,  1505910931000, 1277164609000, 776157290000,
       1048000710000, 660705326000,  505691714000,  961828890000,
       537569964000,  1647153315000, 1108509062000, 520390042000,
       1562168178000, 228138641000,  216429757000,  144986243000,
       180040370000,  110553666000,  1454402814000, 1486942531000,
       997177933000,  871037064000,  308541818000,  711722272000,
       986972089000,  215701966000,  152925788000,  366321659000,
       113202996000,  1247813013000, 546682678000,  1037620761000,
       1259331611000, 346821716000,  448349379000,  856678905000,
       1588465889000, 27989033000,   1039987365000, 38011096000,
       548543874000,  1404831961000, 975542822000,  959676592000,
       1608754451000, 267101901000,  213545088000,  411402585000,
       1289797454000, 719489171000,  1368220405000, 1616706430000,
       828203515000,  1570089093000, 1628274222000, 485133624000,
       27427879000,   990773112000,  1459971704000, 1028681149000,
       739003497000,  1118473283000, 1565971490000, 1068799967000,
       1146617959000, 1618569154000, 665063458000,  1097264699000,
       508254755000,  854529091000,  1219877122000, 1242473274000,
       994825085000,  139569085000,  1526186858000, 1171040845000,
       818924697000,  821791397000,  348825393000,  800751100000,
       993949416000,  1606702195000, 204924001000,  316434687000,
       1161599775000, 1556431645000, 1118134980000, 370288917000,
       68683355000,   194139295000,  805980244000,  818881098000,
       360451089000,  1208001474000, 52798147000,   598388005000,
       1697617591000, 754341415000,  359118509000,  1050351834000,
       669223073000,  593933215000,  376515894000,  1234323405000,
       636220801000,  844276664000,  468877292000,  1231620784000,
       1569192080000, 1134676627000, 1544360014000, 1276701796000,
       1589856582000, 814313893000,  1680863679000, 739917942000,
       765373060000,  1053592948000, 591298237000,  152894038000,
       122274444000,  623608432000,  1339595245000, 572295711000,
       1321711446000, 590461652000,  1664387047000, 428889160000,
       1377810728000, 278253551000,  961302697000,  1680351042000,
       432479133000,  305100482000,  121681876000,  162676687000,
       1386185849000, 296454414000,  1085901027000, 476072718000,
       1440824721000, 361407928000,  634185558000,  1053217842000,
       690789996000,  594467035000,  479000962000,  1370697389000,
       534973573000,  573382020000,  92837967000,   1316242771000,
       1230911977000, 493448656000,  996693522000,  624579281000,
       1183702549000, 1485447212000, 1018416963000, 1697783451000,
       565498027000,  303127780000,  663167997000,  1452032688000,
       85813656000,   1240964020000, 1017695886000, 709395319000,
       1685280659000, 90614567000,   80436851000,   238123582000,
       952174127000,  1233525955000, 581688080000,  659899384000,
       1689732617000, 652322190000,  1110302747000, 1118499653000,
       613260577000,  532033743000,  1574560991000, 712631959000,
       1557403197000, 1192713312000, 1301296158000, 244186605000,
       1044350190000, 181831942000,  1290912745000, 822040398000,
       1304588205000, 1662835346000, 1515325990000, 1571678450000,
       1614670680000, 36245749000,   604608314000,  994733406000,
       457432767000,  928114984000,  1301598474000, 1401630896000,
       333023883000,  536581422000,  1492901206000, 1073520903000,
       867502031000,  139580427000,  1020543630000, 548771218000,
       14093829000,   355143596000,  1072109805000, 1666000817000,
       948929112000,  795580571000,  845239387000,  1457143076000,
       1131797345000, 689576653000,  718274990000,  58358977000,
       189340222000,  1020981950000, 56573081000,   1170279140000,
       341385133000,  1016435808000, 392540472000,  1473363502000,
       1503271797000, 742883730000,  1126268774000, 102760974000,
       6724123000,    1128844321000, 364765074000,  640712296000,
       1271970136000, 1376912164000, 904079677000,  155738804000,
       938159854000,  754081924000,  170977612000,  1245648272000,
       594199549000,  93928477000,   383693436000,  49327638000,
       1240835868000, 1472760961000, 810811291000,  39680907000,
       1306166790000, 1132650540000, 1074308364000, 264514521000,
       122149830000,  502527729000,  789859647000,  1257340037000,
       902424944000,  495109640000,  641843161000,  814744535000,
       1654231877000, 1093485783000, 256026031000,  1646443316000,
       864493601000,  888890381000,  313820154000,  140549411000,
       264514521000,  14093829000,   814744535000,  711313738000,
       1459971704000, 1304588205000, 348825393000,  952174127000,
       1290912745000, 139569085000,  376515894000,  1290912745000,
       1526186858000, 1473363502000, 814313893000,  1565971490000,
       1664387047000, 194139295000,  1208001474000, 1376912164000,
       624579281000,  814744535000,  1028681149000, 1368220405000,
       305100482000,  1618569154000, 1128844321000, 593933215000,
       392540472000,  1557403197000, 1097264699000, 623608432000,
       392540472000,  1557403197000, 814744535000,  1017695886000,
       1503271797000, 305100482000,  1588465889000, 1530268049000,
       305100482000,  1557403197000, 93928477000,   476072718000,
       1569192080000, 493448656000,  1440824721000, 1662835346000,
       305100482000,  305100482000,  1485447212000, 1072109805000,
       1108509062000, 256026031000,  624579281000,  789859647000,
       181831942000,  181831942000,  1440824721000, 1503271797000,
       994825085000,  1037620761000, 123780566000,  411402585000,
       376515894000,  1183702549000, 110553666000,  1068799967000,
       1565971490000, 256026031000,  110553666000,  536581422000,
       1662835346000, 1321711446000, 573382020000,  548771218000,
       845239387000,  1028681149000, 1473363502000, 476072718000,
       244186605000,  565498027000,  1473363502000, 6724123000,
       1108509062000, 355143596000,  1277164609000, 1530268049000,
       1473363502000, 495109640000,  536581422000,  14093829000,
       624579281000,  1565971490000, 85813656000,   1440824721000,
       14093829000,   155738804000,  1230911977000, 613260577000,
       493448656000,  905746043000,  1662835346000, 613260577000,
       1085901027000, 1037620761000, 1085901027000, 814744535000,
       1697617591000, 1053217842000, 1588465889000, 1459971704000,
       305100482000,  313820154000,  1680863679000, 1074308364000,
       1321711446000, 659899384000,  613260577000,  162676687000,
       1685280659000, 1289797454000, 123780566000,  264514521000,
       1370697389000, 457432767000,  1680351042000, 110553666000,
       170977612000,  14093829000,   1697617591000, 457432767000,
       122149830000,  1271970136000, 242395458000,  948929112000,
       194139295000,  1565971490000, 1128844321000, 1233525955000,
       1502828344000, 1454402814000, 361407928000,  1044350190000,
       305100482000,  1037620761000, 1588465889000, 432479133000,
       1259331611000, 493448656000});

  auto timestamp = makeFlatVector<facebook::velox::Timestamp>(
      mills->size(), [mills](auto row) {
        return facebook::velox::Timestamp::fromMillis(mills->valueAt(row));
      });

  testGlobalAgg(timestamp, 0.023, 303);
}
```

The result is 303.
This problem cannot be reproduced when the data set is small. 

### System information

n/a

### Relevant logs

_No response_

### PR DESCRIPTION
**Pull request by 8dukongjian**
Presto uses the millisecond value of Timestamp to calculate the hash value, while Velox uses the Timestamp itself.

The fix is to use the millisecond value of the Timestamp to calculate the hash value.

Fixes #8163

### CODE REVIEW

File: velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp [hunk 1]
```diff
@@ -236,6 +236,130 @@ TEST_F(ApproxDistinctTest, globalAggStrings) {
   testGlobalAgg(values, kFruits.size());
 }
 
+TEST_F(ApproxDistinctTest, globalAggTimeStamp) {
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp:239]: I wonder if there is a way to simplify the test. The fix seems rather generic, hence, I hope the test can be simplified.

Comment (mbasmanova) [reply velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp:239]: For example, 

```
TEST_F(ApproxDistinctTest, globalAggTimeStamp) {
  auto data = makeFlatVector<Timestamp>(
      1'000, [](auto row) { return Timestamp::fromMillis(row); });
  testGlobalAgg(data, 1010);
}
```

Without the fix this test fails:

```
[ RUN      ] ApproxDistinctTest.globalAggTimeStamp
fbcode/velox/exec/tests/utils/QueryAssertions.cpp:1104: Failure
Failed
Expected 1, got 1
1 extra rows, 1 missing rows
1 of extra rows:
	1006

1 of missing rows:
	1010
```

In Presto,

```
presto> select approx_distinct(date_add('millisecond', x, from_unixtime(0))) from unnest(sequence(0, 999)) as t(x);
 _col0
-------
  1010
(1 row)
```

**Review by mbasmanova — COMMENTED** [review]
@8dukongjian Looks good, but let's simplify the test. I made a suggestion inline.

File: velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp (hunk 1 above)
Comment (8dukongjian) [reply velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp:239]: Thanks, updated

**Comment by 8dukongjian** [comment]
> @8dukongjian Looks good, but let's simplify the test. I made a suggestion inline.

Thanks for your suggestion, updated

File: velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp [hunk 2]
```diff
@@ -236,6 +236,12 @@ TEST_F(ApproxDistinctTest, globalAggStrings) {
...
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/tests/ApproxDistinctTest.cpp:239]: nit: globalAggTimeStamp -> globalAggTimestamp

**Review by mbasmanova — APPROVED** [review]
Thanks!

**Comment by mbasmanova** [comment]
@8dukongjian Getting an error: "This diff is either too old, or sits on an old base revision, making it unsafe to land - try rebasing first." Would you rebase?

**Comment by 8dukongjian** [comment]
@mbasmanova Updated, is this okay?

**Comment by 8dukongjian** [comment]
@mbasmanova Tests failed on CircleCI,  the log shows that some test timed out. Need to retry?
```
The following tests FAILED:
        296 - velox_exec_test (Timeout)
Errors while running CTest

Exited with code exit status 8
```

**Comment by mbasmanova** [comment]
@8dukongjian Is this the same failure as https://github.com/facebookincubator/velox/issues/8252 ?

**Pull request closed without merging**